Lib/_pylong.py

```diff
@@ -277,44 +295,46 @@ def inner(n, w):
 
     # How many base 256 digits are needed?. Mathematically, exactly
     # floor(log256(int(s))) + 1. There is no cheap way to compute this.
-    # But we can get an upper buond, and that's necessary for our error
+    # But we can get an upper bound, and that's necessary for our error
     # analysis to make sense. int(s) < 10**len(s), so the log needed is
     # < log256(10**len(s)) = len(s) * log256(10). However, using
     # finite-precision floating point for this, it's possible that the
     # computed value is a little less than the true value. If the true
     # value is at - or a little higher than - an integer, we can get an
     # off-by-1 error too low. So we add 2 instead of 1 if chopping lost
     # a fraction > 0.9.
-    log_ub = len(s) * _LOG_10_BASE_256
+
+    # The "WASI" test platfrom can complain about `len(s)` if it's too
+    # large to fit in its idea of "an index-sized integer".
+    lenS = s.__len__()
```

Comment:
I think that it is a fair game to allow this to fail on WASI. In any case a ValueError is raised for too large input on other platforms. It will just be different (but perhaps close) limit on WASI, and possibly with different error type.

Reply:
I'm strongly inclined not to change this again. It passes now, and the workarounds were easy in the end. Indeed, I'd rather remove the test entirely than waste more time on "WASI" :wink:.